src/java/org/jivesoftware/openfire/plugin/rest/utils/LoggingUtils.java

```diff
@@ -0,0 +1,122 @@
+
+    public enum AuditEvent {
+        //Clustering
+        CLUSTERING_GET_STATUS,
+        CLUSTERING_GET_NODES,
+        CLUSTERING_GET_NODE,
+
+        //Groups
+        GROUPS_LIST,
+        GROUPS_GET_BY_NAME,
+        GROUPS_CREATE,
+        GROUPS_UPDATE_BY_NAME,
+        GROUPS_DELETE,
+
+        //JustMarried
+        USER_CHANGE_NAME,
+
+        //Messages
+        MESSAGE_BROADCAST,
+
+        //Message Archive
+        MESSAGE_ARCHIVE_UNREAD_COUNT,
+
+        //MUC
+        // - MUC Affiliations
+        MUC_LIST_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_REPLACE_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_REMOVE_AFFILIATED_USER_OR_GROUP_FOR_AFFILIATION,
+        MUC_ADD_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_ADMIN,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_MEMBER,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_OUTCAST,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_OWNER,
+        // - MUC Rooms
+        MUC_LIST_ROOMS,
+        MUC_GET_ROOM,
+        MUC_DELETE_ROOM,
+        MUC_CREATE_ROOM,
+        MUC_UPDATE_ROOM,
+        MUC_GET_PARTICIPANT_LIST,
+        MUC_GET_OCCUPANT_LIST,
+        MUC_GET_ROOM_HISTORY,
+        MUC_INVITE_USER,
+        ;
+    }
+
+    public static void auditEvent(AuditEvent event){
+        auditEvent(event, "");
+    }
+
+    public static void auditEvent(AuditEvent event, Object... parameters){
+        if (JiveGlobals.getBooleanProperty(RESTServicePlugin.SERVICE_LOGGING_ENABLED, false)) {
+            String parameterString = parseParameters(parameters);
+            String logMessage = "Event: " + event;
+            logMessage += " - ";
+            logMessage += "Parameters: + " + parameterString;
+            logMessage += " - ";
+            logMessage += "Caller: " + getCaller();
+            AUDIT_LOG.info(logMessage);
+        };
+    }
+
+    private static String parseParameters(Object[] parameters) {
+        //TODO: Does this belong here?
+        ArrayList<String> parsed = new ArrayList<>();
+        for (Object obj: parameters) {
+            if(obj == null){
+                parsed.add("null");
+            } else {
+                try {
+                    parsed.add(obj.toString());
+                } catch (Exception e) {
+                    parsed.add("unparseable");
+                }
+            }
+        }
+        return parsed.toString();
+    }
+
+    /*
+    * Returns the name and method of the calling class.
+    */
+    private static String getCaller() {
+        try {
+            StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+            for (StackTraceElement element : stackTrace) {
+                if(element.getClassName().equals(LoggingUtils.class.getName())){
+                    continue;
+                }
+                return element.getClassName() + "." + element.getMethodName();
+            }
+        } catch (Exception ignored) {}
```

Comment:
I don't think exceptions should ever be ignored. They are either supposed to be not possible (in which case they should be logged to WARN or ERROR when they _do_ get triggered), or be logged to something like TRACE or DEBUG.

Reply:
This could return "Unknown" instead. This probably can't fail, and is only there through an abundance of caution that logging an event should never disrupt the event.

Reply:
I'm fine with not processing the exception, but if you say that it probably can't fail, that's all the more reason to log this very verbosely in case it does pop up. It's one of my pet peeves (after hunting down to many bugs hidden by this): never have empty 'catch' blocks.